ndcube/extra_coords/table_coord.py

```diff
@@ -10,12 +10,71 @@
 from astropy.coordinates import SkyCoord
 from astropy.modeling import models
 from astropy.modeling.models import tabular_model
+from astropy.modeling.tabular import _Tabular
 from astropy.time import Time
 from astropy.wcs.wcsapi.wrappers.sliced_wcs import combine_slices, sanitize_slices
 
 __all__ = ['TimeTableCoordinate', 'SkyCoordTableCoordinate', 'QuantityTableCoordinate']
 
 
+class Length1Tabular(_Tabular):
+    _input_units_allow_dimensionless = True
+    _has_inverse_bounding_box = True
+    _separable = True
+
+    n_inputs = 1
+    n_outputs = 1
+
+    lookup_table = np.zeros([1])
+    points = np.zeros([1])
+
+    def __init__(self, points=None, lookup_table=None, pixel_width=None,
+                 method='linear', bounds_error=True, fill_value=np.nan, **kwargs):
+        if len(lookup_table) != 1:
+            raise ValueError("lookup_table must have length 1.")
+        super().__init__(points=points, lookup_table=lookup_table, method=method,
+                         bounds_error=bounds_error, fill_value=fill_value, **kwargs)
+        self._pixel_width = pixel_width  # Width of pixel in world units.
+
+    def evaluate(self, x):
+        output = np.full(x.shape, self.fill_value)
+        x_unit = self.input_units['x']
+        output[np.logical_and(x >= -0.5*x_unit, x < 0.5*x_unit)] = self.lookup_table[0].value
```

Comment:
Should this not use `pixel_width`?

Reply:
No, `pixel_width` is effectively CDELT.  So it's in real world coordinates and needs to be defined by the user.  Here we need the pixel width in pixel units which is always defined as -0.5 - 0.5 pix.

Reply:
Should this also have it's own parameter then? to match what is currently `pixel_width`, we are just assuming the point has a width of 1, it might not (in the general sense).